docs/how_to/openedx/release_upgrades.md

```diff
@@ -0,0 +1,139 @@
+     mitx_staging = DeploymentEnvRelease(
+         deployment_name="mitx-staging",
+         env_release_map=[
+-            EnvRelease("CI", OpenEdxSupportedRelease["olive"]),
++            EnvRelease("CI", OpenEdxSupportedRelease["palm"]),
+             EnvRelease("QA", OpenEdxSupportedRelease["olive"]),
+             EnvRelease("Production", OpenEdxSupportedRelease["olive"]),
+         ],
+```
+
+Because Palm is a new release for these deployments we also need to add a `palm` key to
+the `ReleaseMap` dictionary that contains the applications that are associated with
+those deployments and the appropriate `OpenEdxApplicationVersion` records for that
+deployment.
+
+```diff
+@@ -61,6 +61,122 @@ ReleaseMap: dict[
+     OpenEdxSupportedRelease,
+     dict[OpenEdxDeploymentName, list[OpenEdxApplicationVersion]],
+ ] = {
++    "palm": {
++        "mitx": [
++            OpenEdxApplicationVersion(
++                application="edx-platform",  # type: ignore
++                application_type="IDA",
++                release="palm",
++                branch_override="mitx/palm",
++                origin_override="https://github.com/mitodl/edx-platform",
++            ),
+             ...
++        ],
++        "mitx-staging": [
++            OpenEdxApplicationVersion(
++                application="edx-platform",  # type: ignore
++                application_type="IDA",
++                release="palm",
++                branch_override="mitx/palm",
++                origin_override="https://github.com/mitodl/edx-platform",
++            ),
+             ...
++        ],
++    },
+     "olive": {
+         "mitx": [
+             OpenEdxApplicationVersion(
+```
+
+All of the [deployment
+pipelines](https://github.com/mitodl/ol-infrastructure/blob/main/src/concourse/pipelines/open_edx/)
+for these application components are managed by a corresponding `meta` pipeline that
+will automatically update the build and pipeline configuration based on the changed
+version information as soon as it is merged into the `master` branch of
+`ol-infrastructure`.
+
+**Note - TMM 2023-04-14**
+
+- The current configuration of our meta pipelines means that before the updates to the
+  `bridge.settings.openedx.version_matrix` module can be picked up by the `meta`
+  pipelines the `ol-infrastructure` docker image needs to be built and pushed to our
+  registry so that it can be loaded. This means that once the [ol-infrastructure image
+  pipeline](https://cicd.odl.mit.edu/teams/main/pipelines/ol-infrastructure-docker-container)
+  completes it might be necessary to manually trigger the meta pipelines again.
+- Once a new release is deployed to a given environment stage for the first time it may
+  be necessary to manually ensure that all database migrations are run properly. It will
+  be attempted automatically on deployment, but there are often conflicts between the
+  existing database state and the migration logic that require intervention.
```

Comment:
It be helpful to provide additional info on how/where to check for failed migrations and a hint as to how to go about resolving those issues.

Reply:
I agree that this is useful information. I'm going to open an issue to write a how-to on that as a separate document. Once that's written I'll link it to this doc.